parser_test.go

```diff
@@ -1257,6 +1257,38 @@ func TestArrayEach(t *testing.T) {
 	}, "a", "b")
 }
 
+func TestArrayEachNull(t *testing.T) {
+	funcError := func([]byte, ValueType, int, error) { t.Errorf("Could not iterate through array because value is null") }
+
+	type args struct {
+		data []byte
+		cb   func(value []byte, dataType ValueType, offset int, err error)
+		keys []string
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantOffset int
+		wantErr    bool
+	}{
+		{"Array in second key's value with null", args{[]byte(`{"a":{"z":null,"b":[null],"c":null},"d":[]}`), funcError, []string{}}, 1, false},
```

Comment:
formatting nit: break the fields into separate labeled lines to more resemble our other testing, for example
```
{
name: "stuff",
args: args,
etc....
}
```